libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/utils/extensions/Mappers.kt

```diff
@@ -16,16 +39,158 @@ fun WalkingOptionsNavigation.mapToWalkingOptions(): WalkingOptions = WalkingOpti
     .build()
 
 fun Route.mapToDirectionsRoute(): DirectionsRoute {
-    val duration = duration.toDouble()
-    val legs = legs?.let { it as List<RouteLeg> }
-
-    return DirectionsRoute.builder()
+    return DirectionsRoute.builder() // FIXME route index
         .distance(distance)
-        .duration(duration)
+        .duration(duration.toDouble())
         .geometry(geometry)
         .weight(weight)
         .weightName(weightName)
         .voiceLanguage(voiceLanguage)
-        .legs(legs)
+        .legs(legs?.map(RouteLegNavigation::mapToRouteLeg))
+        .routeOptions(routeOptions?.mapToRouteOptions())
         .build()
 }
+
+fun RouteOptionsNavigation.mapToRouteOptions() =
+    RouteOptions.builder()
+        .apply {
+            walkingOptions?.let { walkingOptions(it.mapToWalkingOptions()) }
+            accessToken?.let { accessToken(it) }
+            exclude?.let { exclude(it) }
+            requestUuid?.let { requestUuid(it) }
+        }
+        .profile(profile)
+        .user(user)
+        .baseUrl(baseUrl)
+        .alternatives(alternatives)
+        .annotations(annotations)
+        .approaches(approaches)!!
```

Comment:
We should fix (remove `@Nullable`) annotations in `mapbox-java` e.g. https://github.com/mapbox/mapbox-java/blob/9bd6158292374fc6dd5f7eba33e7586ca1d76177/services-directions/src/main/java/com/mapbox/api/directions/v5/models/RouteOptions.java#L568 `Builder` cannot be `null` to avoid unnecessary `!!`. The same thing happens with another methods below.

Reply:
should we make it self or create a ticket for Direction API team? what is algorithm for it?

Reply:
Yeah feel free to cut a PR in [`mapbox-java`](https://github.com/mapbox/mapbox-java) fixing this (should be a pretty straightforward one 😛). We're in charge of maintaining the nav-related services, the Directions API team does know nothing about these wrappers 😅